### PR TITLE
[WFLY-7442] Expose UndertowListener via the capability

### PR DESCRIPTION
**Pull request by bstansberry**
@stuartwdouglas if this looks good to you then i'll jump through the JBEAP hoops.

### CODE REVIEW

**Comment by ctomc** [comment]
This doesn't expose UndertowService via capability but adds alias Listener to service.
I don't see what good is this for?

**Comment by bstansberry** [comment]
@ctomc Oops! The commit message/ PR title were wrong; since fixed.

In his WFLY-7442 work Stuart changed the value type provided by ListenerService from itself to the new UndertowListener class. But he didn't update the RuntimeCapability to reflect that. That's the main point of this PR. Plus the service was never being exposed using the ServiceName the capability indicates.

File: undertow/src/main/java/org/wildfly/extension/undertow/ListenerAdd.java [hunk 1]
```diff
@@ -105,7 +105,9 @@ public HttpHandler wrap(HttpHandler handler) {
         serviceBuilder.addDependency(workerServiceName, XnioWorker.class, service.getWorker())
                 .addDependency(socketBindingServiceName, SocketBinding.class, service.getBinding())
                 .addDependency(bufferPoolServiceName, (Injector) service.getBufferPool())
-                .addDependency(UndertowService.SERVER.append(serverName), Server.class, service.getServerService());
+                .addDependency(UndertowService.SERVER.append(serverName), Server.class, service.getServerService())
+                // Include the capability-declared service name as an alias  TODO get rid of the old style name
+                .addAliases(ListenerResourceDefinition.LISTENER_CAPABILITY.getCapabilityServiceName(name));
```
Comment (ctomc) [inline undertow/src/main/java/org/wildfly/extension/undertow/ListenerAdd.java]: I would do this bit differently.
in line 73 we have: 
`final ServiceName listenerServiceName = UndertowService.listenerName(name);`
which should be changed to 
`final ServiceName listenerServiceName = ListenerResourceDefinition.LISTENER_CAPABILITY.getCapabilityServiceName(name);` 

Unless your TODO was meant to do just that :-)

Comment (ctomc) [inline undertow/src/main/java/org/wildfly/extension/undertow/ListenerAdd.java]: nevermind my comment, I get it why you used alias.
I should probably push my branch that has more than just this changes on it. 
This is why i am confusing what is "done" and what is done.

**Comment by bstansberry** [comment]
retest this please

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**